src/object_schema.cpp

```diff
@@ -214,6 +240,62 @@ static void validate_property(Schema const& schema,
 
 void ObjectSchema::validate(Schema const& schema, std::vector<ObjectSchemaValidationException>& exceptions) const
 {
+    std::vector<StringData> public_property_names;
+    std::vector<StringData> internal_property_names;
+    internal_property_names.reserve(persisted_properties.size() + computed_properties.size());
+    auto gather_names = [&](auto const &properties) {
+        for (auto const &prop : properties) {
+            internal_property_names.push_back(prop.name);
+            if (!prop.public_name.empty())
+                public_property_names.push_back(prop.public_name);
+        }
+    };
+    gather_names(persisted_properties);
+    gather_names(computed_properties);
+    std::sort(public_property_names.begin(), public_property_names.end());
+    std::sort(internal_property_names.begin(), internal_property_names.end());
+
+    // Check that property names and aliases are unique
+    auto for_each_duplicate = [](auto &&container, auto &&fn) {
+        auto end = container.end();
+        for (auto it = std::adjacent_find(container.begin(), end); it != end; it = std::adjacent_find(it + 2, end))
```

Comment:
you'll get multiple calls to `fn` if there are more than 4 duplicates...
Have you considered using `std::unique`?

Reply:
`std::unique` doesn't tell you what the duplicates were. Maybe something like `for (auto it = std::adjacent_find(container.begin(), end); it != end; it = std::adjacent_find(std::find_if(it + 2, end, [&](auto&& v) { v != *it; }, end))`

Reply:
I added unit tests for this. The current implementation does the correct thing also, if there are more than 4 duplicates.